### PR TITLE
VideoPress: re-write token bridge lib

### PR DESCRIPTION
**Pull request by retrofox**
This PR re-writes the token bridge library in order to know its functionality:
* Use TypeScript
* Add debug() lines

Fixes #

## Proposed changes:

* VideoPress: re-write token bridge lib

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to the block editor
* Open the network tab
* filter by `token-bridge`
* Confirm the app loads the script a few times. We need to polish that, btw. Follow-up.
* Open the console and filter by `token-bridge` too
* Hard refresh
* Add/edit a video block
* Change the video privacy
* Check how the bridge listens and responds when the client requests the token
* Check with `v5` and `v6`. In the block editor context and the front-end.

https://user-images.githubusercontent.com/77539/215863982-6fae4190-509c-4535-bb35-41bd7ea6fb08.mov

### CODE REVIEW

File: projects/packages/videopress/src/client/lib/token-bridge/index.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import debugFactory from 'debug';
+/**
+ * Types
+ */
+import getMediaToken from '../get-media-token';
+import type { VideoGUID } from '../../block-editor/blocks/video/types';
+
+const debug = debugFactory( 'videopress:token-bridge' );
+
+type videopressAjaxPostMessageEvenProps = {
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/lib/token-bridge/index.ts]: There is a missing `t` on `Event`, and we can capitalize the first letter to keep consistency:

```suggestion
type VideopressAjaxPostMessageEventProps = {
```

File: projects/packages/videopress/src/client/lib/token-bridge/index.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+};
+
+/**
+ * Function handler to dialog with the client
+ * (toquen requester) and the app
+ * to provide a JWT on demand.
+ *
+ * @param {object} event - The event object
+ */
+export async function tokenBrigeHandler(
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/lib/token-bridge/index.ts]:
```suggestion
export async function tokenBridgeHandler(
```

File: projects/packages/videopress/src/client/lib/token-bridge/index.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+type TokenBrigeEventProps = {
+	event: 'videopress_token_request';
+	guid: VideoGUID;
+	requestId: string;
+	origin: Origin;
+};
+
+/**
+ * Function handler to dialog with the client
+ * (toquen requester) and the app
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/lib/token-bridge/index.ts]:
```suggestion
 * (token requester) and the app
```

**Comment by lhkowalski** [comment]
Code looks really good. Left some comments related to typos, and now I will test the change following the instructions.

**Comment by lhkowalski** [comment]
While testing it on a Jetpack site, I got the js file request a few times as well, and the debug messages are poping:

Screen Shot 2023-02-01 at 11 40 10

But I also got this error:

Screen Shot 2023-02-01 at 11 43 30

I am using Chrome and my local dev env. I will lunch a fresh JN to test it again, just in case.

**Comment by retrofox** [comment]
> While testing it on a Jetpack site, I got the js file request a few times as well, and the debug messages are poping:
> 
> Screen Shot 2023-02-01 at 11 40 10

If I'm right, this issue already happens in trunk. Something that I'd like to address in a follow-up PR.

**Comment by lhkowalski** [comment]
Tested on a fresh site, it works as described on the testing instructions:

Screen Shot 2023-02-01 at 12 00 54

I will check with v5, as well as inside WPCOM.

**Comment by lhkowalski** [comment]
Working for v5 as well:

Screen Shot 2023-02-01 at 12 03 49

**Comment by lhkowalski** [comment]
The bridge is not loaded on WPCOM, but that is the expected behavior.

File: projects/packages/videopress/src/client/lib/token-bridge/index.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	} );
+
+	if ( ! tokenData?.token ) {
+		debug( '(%s) Error getting token', context );
+		tokenRequester.postMessage(
+			{
+				event: 'videopress_token_error',
+				guid: event.data.guid,
+				requestId,
+			} as videopressAjaxPostMessageEvenProps,
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/lib/token-bridge/index.ts]:
```suggestion
			} as VideopressAjaxPostMessageEventProps,
```

File: projects/packages/videopress/src/client/lib/token-bridge/index.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+	}
+
+	debug( '(%s) sending token', context );
+	tokenRequester.postMessage(
+		{
+			event: 'videopress_token_received',
+			guid: guid,
+			jwt: tokenData.token,
+			requestId,
+		} as videopressAjaxPostMessageEvenProps,
```
Comment (lhkowalski) [inline projects/packages/videopress/src/client/lib/token-bridge/index.ts]:
```suggestion
		} as VideopressAjaxPostMessageEventProps,
```

**Review by lhkowalski — DISMISSED** [review]
Code looks good now, and the tests are working as expected. Let's move forward!

**Review by lhkowalski — APPROVED** [review]
Nice catch on the last bug fix.

**Pull request merged**